fe1-web/src/features/rollCall/components/__tests__/EventRollCall.test.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react-native';
+import STRINGS from 'resources/strings';
+import { Hash, Timestamp } from 'core/objects';
+import { useNavigation } from '@react-navigation/native';
+
+import * as reactRedux from 'react-redux';
+import { mockLao } from '__tests__/utils/TestUtils';
+import { requestOpenRollCall, requestReopenRollCall } from 'features/rollCall/network';
+import EventRollCall from '../EventRollCall';
+import { EventTypeRollCall, RollCall, RollCallStatus } from '../../objects';
+
+const ID = new Hash('rollCallId');
+const NAME = 'myRollCall';
+const LOCATION = 'location';
+const TIMESTAMP_START = new Timestamp(1620255600);
+const TIMESTAMP_END = new Timestamp(1620357600);
+const ATTENDEES = ['attendee1', 'attendee2'];
+const rollCallState: any = (mockStatus: RollCallStatus) => {
+  return {
+    id: ID.valueOf(),
+    eventType: EventTypeRollCall,
+    start: TIMESTAMP_START.valueOf(),
+    name: NAME,
+    location: LOCATION,
+    creation: TIMESTAMP_START.valueOf(),
+    proposedStart: TIMESTAMP_START.valueOf(),
+    proposedEnd: TIMESTAMP_END.valueOf(),
+    status: mockStatus,
+    attendees: ATTENDEES,
+    idAlias: mockStatus === RollCallStatus.CREATED ? undefined : ID.valueOf(),
+  };
+};
+
+const mockRollCallClosed = RollCall.fromState(rollCallState(RollCallStatus.CLOSED));
+const mockRollCallCreated = RollCall.fromState(rollCallState(RollCallStatus.CREATED));
+const mockRollCallOpened = RollCall.fromState(rollCallState(RollCallStatus.OPENED));
+
+const mockRenderRollCall = (rollCall: RollCall, isOrganizer: boolean) => {
+  return render(<EventRollCall event={rollCall} isOrganizer={isOrganizer} />);
+};
+
+jest.mock('@react-navigation/native');
+jest.mock('features/rollCall/network');
+const mockNavigate = jest.fn();
+
+let mockSelector: jest.SpyInstance<
+  unknown,
+  [
+    selector: (state: unknown) => unknown,
+    equalityFn?: ((left: unknown, right: unknown) => boolean) | undefined,
+  ]
+>;
+
+beforeEach(() => {
+  jest.resetAllMocks();
+  mockSelector = jest.spyOn(reactRedux, 'useSelector').mockReturnValueOnce(mockLao);
+  (useNavigation as jest.Mock).mockImplementation(() => {
+    return {
+      navigate: mockNavigate,
+    };
+  });
+  (requestOpenRollCall as jest.Mock).mockImplementation(() => Promise.resolve());
+  (requestReopenRollCall as jest.Mock).mockImplementation(() => Promise.resolve());
```

Comment:
I don't think you need to declare these. We already have a file [there](https://github.com/dedis/popstellar/blob/master/fe1-web/src/features/rollCall/network/__mocks__/RollCallMessageApi.ts) where you can add these mock requests. Since you've already written `jest.mock('features/rollCall/network')`, the mock should be correctly imported.

Reply:
I need to declare them because `EventRollCall` calls a catch on them which does not work when undefined, I used [this](https://github.com/dedis/popstellar/blob/master/fe1-web/src/features/rollCall/screens/__tests__/RollCallOpened.test.tsx#L56) as a template because I wasn't sure about the cleanest way of doing it. Also what would be the reason to use the [external mock request file](https://github.com/dedis/popstellar/blob/master/fe1-web/src/features/rollCall/network/__mocks__/RollCallMessageApi.ts) if the goal is to create a mock function returning a promise, which is a one liner ?

Reply:
Sure, it works fine that way. The purpose if doing it with an external file is that you're not going to specify the function you want to mock in each file where you're going to use it. You just have to write `jest.mock('filePath')` and import the function that you want to use directly. It might be unclear, but we can talk about it in the TypeScript meeting if needed.